src/ja/common/message/base.py

```diff
@@ -144,13 +144,16 @@ class Response(Message):
     A base class for messages which are sent as a response to Commands. They
     indicate the result of the action on the remote component.
     """
-    def __init__(self, result_string: str, is_success: bool):
+    def __init__(self, result_string: str, is_success: bool, uid: str = None):
```

Comment:
You should use a subclass here, not add a field which is not relevant for all responses.

Reply:
I am intentionally not using sub-classes of Response.
Due to the Liskov substitution principle it is not easily possible to return a more specific class when a method is overridden.
I think the implementation will be cleaner this way.

Reply:
What? LSP states that you can return with stronger guarantees. You cannot take a more specific argument.

Reply:
I think you're right in that I used the LSP incorrectly.
The specific problem I'm having:
In SSHConnection I want to implement a method like `send_command(command: Command) -> Response`.
If only some sub-classes of Response had the uid property I would need to use casts to access the property.

Reply:
I think a cast should be fine in this case, but whatever, its not critical.